fuel-vm/src/checked_transaction.rs

```diff
@@ -121,6 +125,68 @@ impl<Tx: IntoChecked> Checked<Tx> {
     }
 }
 
+/// Transaction that has checks for all dynamic values, e.g. `gas_price`
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct Ready<Tx: IntoChecked> {
+    gas_price: Word,
+    transaction: Tx,
+    metadata: Tx::Metadata,
+    checks_bitmask: Checks,
+}
+
+impl<Tx: IntoChecked> Ready<Tx> {
+    /// Consume and decompose components of the `Immutable` transaction.
+    pub fn decompose(self) -> (Word, Tx, Tx::Metadata, Checks) {
```

Comment:
I think it makes sense to return `(Word, Checked<Tx>)`. It will give more flexibility to the consumer of this method. They can decide do they need to break down it more or not without losing checkness=)